File: C++/Programming Basics with C++/09. Exercise - While-loop/simple-tasks/OldBooks.cpp
#include<iostream>
#include <string>

using namespace std;

int main() {
    string searchingBook;
    getline(cin, searchingBook);

    int capacity;
    cin >> capacity;
    cin.ignore();

    int counter = 0;
    bool isFound = false;
    while (counter < capacity) {
        string currentBook;
        getline(cin, currentBook);

        if (currentBook == searchingBook) {
            cout << "You checked " << counter << " books and found it." << endl;
            isFound = true;
            break;
        }

        counter++;
    }

    if (!isFound) {
        cout << "The book you search is not here!" << endl;
        cout << "You checked " << counter << " books." << endl;
    }
    return 0;
}

File: C#/Programming Basics with C#/09. While-Loop - Exercise/C02. Old Books/OldBooks.cs
using System;

class OldBooks
{
    static void Main()
    {
        string serchedbook = Console.ReadLine();
        int capacity = int.Parse(Console.ReadLine());
        string checkedBook = " ";
        int counter = 0;
        while (serchedbook != checkedBook && counter < capacity)
        {
            checkedBook = Console.ReadLine();
            ++counter;
        }
        if (serchedbook == checkedBook)
        {
            Console.WriteLine($"You checked {counter - 1} books and found it.");
        }
        else
        {
            Console.WriteLine("The book you search is not here!");
            Console.WriteLine($"You checked {counter} books.");
        }
    }
}
